docs/guides/hardhat-walkthrough/deploy-custom-lsp7.md

```diff
@@ -0,0 +1,158 @@
+---
+sidebar_label: Deploy our custom LSP7 contract
+sidebar_position: 3
+---
+
+# Deploy your contract
+
+Following the previous guide ("Create a custom LSP7 contract"), we are now ready to deploy it on the LUKSO Testnet network!
+
+## Deploy the contract on LUKSO Testnet
+
+In order to deploy the contract, we will have to update the `hardhat.config.ts` and create a deploy script. Let's go!
+
+### Update hardhat config
+
+Jump in the `hardhat.config.ts` and update the file with this:
+
+```ts title="Update hardhat config"
+import { HardhatUserConfig } from 'hardhat/config';
+import { config as LoadEnv } from 'dotenv';
+import '@nomicfoundation/hardhat-toolbox';
+
+LoadEnv();
+
+const config: HardhatUserConfig = {
+  solidity: '0.8.9',
+  networks: {
+    luksoTestnet: {
+      url: 'https://rpc.testnet.lukso.network',
+      chainId: 4201,
+      accounts: [process.env.PRIVATE_KEY as string],
+    },
+  },
+};
+
+export default config;
+```
+
+### Create a deploy script
+
+We will create a script to deploy the smart contract to the LUKSO Testnet network. You can either use a regular EOA (Externally Owned Account) or a Universal Profile. Let's see those 2 possibilities below.
+
+#### Deploy using an Universal Profile (Recommended)
+
+In this chapter, we are going to deploy our contract using our Universal Profile. First thing is to [Install the UP browser extension](https://docs.lukso.tech/guides/browser-extension/install-browser-extension). Once installed, we will retrieved the needed information:
+
+- Clic on the extension
+- Clic on the cogwheel ⚙️ at the top right corner, then select "reveal private keys"
+- Enter your password
+- Scroll down and copy the `privateKey` field to your `.env` file in `PRIVATE_KEY`
+- Copy the `address` field to your `.env` file in `UP_ADDR`
+
+:::note
+The `privateKey` coming from your UP extension is the private key of the EOA that controls your UP. You can find the associated address in the extension if you click on the controller tab > UP Extension. This address will need to be funded using the [Testnet Faucet](https://faucet.testnet.lukso.network/).
+:::
+
+Now that we are all set up, we will create the script that will deploy the contract as your UniversalProfile. In order to do so, we will:
+
+- Create a wallet instance with our private key (the `signer`)
+- Load the associated UP
+- Get the bytecode of our contract
+- use `staticCall` method to get the address of the contract
+- deploy the contract
+
+Go in the `scripts/` folder and create a file named `deployUP.ts` with the following content:
```

Comment:
Maybe we can also add this in a repo like here:

https://github.com/lukso-network/lukso-playground

Reply:
done